presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -213,13 +213,14 @@ public QueryId createQueryId()
                     retryCount,
                     selectionContext.getResourceGroupId(),
                     queryType,
-                    warningCollector);
+                    warningCollector,
+                    (dq) -> resourceGroupManager.submit(preparedQuery.getStatement(), dq, selectionContext, queryExecutor));
```

Comment:
`dq` -> `query`

Reply:
I wanted to do that myself - but we already have `dispatchQuery` and `query` variables in this context